File: 3d_printer.pl
my ($X, $Y, $Z) = split(" ", <STDIN>);
my @S;

for (my $z = 0; $z < $Z; $z++) {
    for (my $x = 0; $x < $X; $x++) {
        chomp(my $line = <STDIN>);
        push @{$S[$z]}, $line;
    }
    <STDIN>;
}

for (my $z = $Z - 1; $z >= 0; $z--) {
    for (my $y = 0; $y < $Y; $y++) {
        my $c = '.';
        for (my $x = 0; $x < $X; $x++) {
            if (substr($S[$z][$x], $y, 1) eq '#') {
                $c = '#';
                last;
            }
        }
        print $c;
    }
    print "\n";
}

File: 3d_printer.scala
import scala.io.StdIn._

object Main extends App {
    val Array(x, y, z) = readLine().split(" ").map(_.toInt)
    val S = Array.ofDim[String](z, x)

    for (i <- 0 until z) {
        S(i) = Array.fill(x)(readLine())
        readLine()
    }

    for (i <- z - 1 to 0 by -1) {
        val t = new StringBuilder()
        for (j <- 0 until y) {
            val c = if ((0 until x).exists(k => S(i)(k)(j) == '#')) '#' else '.'
            t.append(c)
        }
        println(t.toString())
    }
}